File: official/docs/python/current/carbon-offset/one-call-buy.py
import easypost


easypost.api_key = "<YOUR_TEST/PRODUCTION_API_KEY>"

to_address = easypost.Address.create(...)
from_address = easypost.Address.create(...)
parcel = easypost.Parcel.create(...)

# Reusing Easypost Objects

shipment = easypost.Shipment.create(
    with_carbon_offset=True,
    carrier_accounts=["ca_..."],
    service="NextDayAir",
    to_address=to_address,
    from_address=from_address,
    parcel=parcel,
)

File: official/docs/ruby/current/carbon-offset/one-call-buy.rb
require 'easypost'
EasyPost.api_key = "<YOUR_TEST/PRODUCTION_API_KEY>"

to_address = EasyPost::Address.create(...)
from_address = EasyPost::Address.create(...)
parcel = EasyPost::Parcel.create(...)
customs_info = EasyPost::CustomsInfo.create(...)

EasyPost::Shipment.create(
  carrier_accounts: [“ca_...”],
  service: “NextDayAir”,
  to_address: to_address,
  from_address: from_address,
  parcel: parcel,
  nil,
  true
)
